### PR TITLE
adding reboot for each stage

### PR DESCRIPTION
**Pull request by okakarpa**
Adding a new Build step (Triggers reboot of a node using ansible) after every build step for each and every stage

Tried different approaches (Declarative/scripted) pipelines for rebooting before build steps but would disconnect from the node and doesn't move further after the reboot.

### CODE REVIEW

**Comment by daniellowell** [comment]
How does adding an additional reboot job affect the return value of the test job? That is, does it return success when successfully rebooting a node, but the test failed?

**Comment by pfultz2** [comment]
Does this require ansible to be installed on the node? How do we setup a node for this? Also, it should probably use the jenkins user and not fpadmin. Does ansible require full sudo privileges? We really should restrict sudo access to just allow rebooting on nodes as its a security risk.

**Comment by okakarpa** [comment]
> Does this require ansible to be installed on the node? How do we setup a node for this? Also, it should probably use the jenkins user and not fpadmin. Does ansible require full sudo privileges? We really should restrict sudo access to just allow rebooting on nodes as its a security risk.

File: Jenkinsfile [hunk 1]
```diff
@@ -203,6 +203,7 @@ pipeline {
                     }
                     steps{
                         buildJob('hcc', flags: '-DCMAKE_BUILD_TYPE=release', cmd: cmd, gpu_arch: "all")
+                        build job: 'reboot-slaves' , parameters: [string(name: 'server', value: "${env.NODE_NAME}"),]
```
[the remark this answers is not in the document]
Comment (okakarpa) [reply Jenkinsfile:206]: should i be adding reboot from **Fast full precision** stage

**Comment by okakarpa** [comment]
> Does this require ansible to be installed on the node? How do we setup a node for this? Also, it should probably use the jenkins user and not fpadmin. Does ansible require full sudo privileges? We really should restrict Sudo access to just allow rebooting on nodes as its a security risk.

Ansible is already installed on the control machine from where the playbooks get executed and since reboot requires sudo privileges I am using fpadmin user

**Comment by pfultz2** [comment]
> Ansible is already installed on the control machine from where the playbooks get executed

Which machine is that? Is there a repo with the playbooks?

> reboot requires sudo privileges I am using fpadmin user

The jenkins user should have sudo privileges to reboot(but it doesnt have all sudo privileges as fpadmin has).

**Comment by okakarpa** [comment]
> > Ansible is already installed on the control machine from where the playbooks get executed
> 
> Which machine is that? Is there a repo with the playbooks?
> 
> > reboot requires sudo privileges I am using fpadmin user
> 
> The jenkins user should have sudo privileges to reboot(but it doesnt have all sudo privileges as fpadmin has).

since I didn't push the playbooks to GitHub repo the pipeline is triggering another Jenkins job which would take care of the reboot, will be working on conditional step for the reboot and push the playbooks to the repository will keep on hold for the existing PR

**Comment by pfultz2** [comment]
> since I didn't push the playbooks to GitHub repo

Do you have a repo already to push these playbooks?

> the pipeline is triggering another Jenkins job which would take care of the reboot,

This seems fine as along as it reports the correct error from jenkins.

> will be working on conditional step for the reboot and push the playbooks to the repository will keep on hold for the existing PR

What is the conditional step for?

**Comment by pfultz2** [comment]
> How does adding an additional reboot job affect the return value of the test job? That is, does it return success when successfully rebooting a node, but the test failed?

The commit statuses are set by the [`githubStatusWrapper`](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/669/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R130) which doesnt include the reboot so this shouldn't be a problem.

**Comment by okakarpa** [comment]
> > since I didn't push the playbooks to GitHub repo
> 
> Do you have a repo already to push these playbooks?
> 
> > the pipeline is triggering another Jenkins job which would take care of the reboot,
> 
> This seems fine as along as it reports the correct error from jenkins.
> 
> > will be working on conditional step for the reboot and push the playbooks to the repository will keep on hold for the existing PR
> 
> What is the conditional step for?

so basically when the build step fails due to driver errors that wouldn't do a reboot of the node writing a conditional step to make sure reboot happens irrespective of the build step to be passed or failed but the end result of the entire stage is always dependent on the build step but not on the reboot

**Comment by okakarpa** [comment]
@pfultz2  should I be adding reboot to all stages or can I skip static analysis stage

**Comment by pfultz2** [comment]
> can I skip static analysis stage

I would just skip this stage.

**Comment by pfultz2** [comment]
> but the end result of the entire stage is always dependent on the build step but not on the reboot

This already happens, as the result of the build stage is reported by `githubStatusWrapper` which doesnt include the reboot.

**Pull request closed without merging**